bpf/lib/icmp6.h

```diff
@@ -355,10 +374,19 @@ static __always_inline int __icmp6_handle_ns(struct __ctx_buff *ctx, int nh_off)
 	cilium_dbg(ctx, DBG_ICMP6_NS, target.p3, target.p4);
 
 	BPF_V6(router, ROUTER_IP);
-	if (ipv6_addrcmp(&target, &router) == 0) {
+	router_target = ipv6_addrcmp(&target, &router);
+
+	if (router_target) {
```

Comment:
Isn't this condition the opposite of what the previous code tested for?

Reply:
Ah.. sorry, it is somehow mistakenly changed.